File: leetsync/1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.go
func checkIfExist(arr []int) bool {
	dict := map[int]bool{}
	for _, v := range arr {
		if dict[2*v] || (dict[v/2] && v%2 == 0) {
			return true
		}
		dict[v] = true
	}
	return false
}

File: leetsync/1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        int n = arr.length;
        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < n; i++) {
            // If 2 * arr[i] is already in the set, return true
            int dble = arr[i] * 2;
            int half = arr[i] / 2;
            if ((set.contains(dble)) || ((arr[i] % 2 == 0) && (set.contains(half)))) {
                return true;
            } else {
                set.add(arr[i]);
            }
        }
        return false;
    }
}